### ISSUE TITLE
[49.8] Starting scenario only loads ally units - no opfor.

### ISSUE THREAD

**Issue by Thom293** [MegaMek/mekhq#3230]
16:41:55,296 INFO  [mekhq.MekHQ] {main}
mekhq.MekHQ.initializeLogging(MekHQ.java:254) - Starting MekHQ v0.49.8-SNAPSHOT
	Build Date: 2022-04-03T01:39:41.469496400
	Release Date: 2022-02-18
	Today: 2022-04-18
	Origin Project: MekHQ
	Java Vendor: AdoptOpenJDK
	Java Version: 11.0.10
	Platform: Windows 10 10.0 (amd64)
	System Locale: en_US
	Total memory available to MekHQ: 4 GB

### Description
Just as title.  And it doesnt even load all of ally units.  Stops after 3.  Tried saving and reloading - did not work.  The 3 tanks of mine are added as bonus from Leadership.  This is first mission on new contract, so not a ghost of them as in the other bug I posted. 

EDIT:  I tried again and let it sit for a while, to see if it was just taking a long time to load.  It did eventually start loading but gave me a you have been disconnected from the server message and closed.

EDIT2:  I rolled another Irregular battle the next week with same lance and deployed them without the leadership bonus extra 3 units and it deployed fine.  So maybe something related to them and/or #3227 ?

I have a custom mech below.  Would not let me upload mtf file.  I changed to txt.  Ill just delete the custom going forward.  Too much of a pain.

### Files
![Screenshot (410)](https://user-images.githubusercontent.com/23645569/163883459-65ebbcaf-0b0c-425f-a3e6-bba1e176b4c6.png)
![Screenshot (411)](https://user-images.githubusercontent.com/23645569/163883490-2e5f9bc4-4f9e-40ab-bfdb-d1abf271973b.png)
[The Bluecoats30620204.cpnx.gz](https://github.com/MegaMek/mekhq/files/8508378/The.Bluecoats30620204.cpnx.gz)
[mekhq.log](https://github.com/MegaMek/mekhq/files/8508381/mekhq.log)
[megamek.log](https://github.com/MegaMek/mekhq/files/8508382/megamek.log)
[Atlas AS7-DLL.txt](https://github.com/MegaMek/mekhq/files/8508385/Atlas.AS7-DLL.txt)

**Comment by JonhyBurner** [comment]
I had something like this happen recently when using a potato laptop, but couldn't repeat it on my regular machine, so I chalked it up to slowness, so no logs. I did had a peek at the time, and it had the same warnings regarding the connection. Similar situation: My forces got loaded, friendly tank lance didn't. When I GM'd a platoon from that lance in the briefing tab to have one less trooper and then re-added him, the game managed to load the lobby properly (forgot to check the logs after that).

**Comment by Thom293** [comment]
Yeah I thought that might be it too, but I have loaded the same size Irregular battles many times.  And this is a gaming desktop.  Its a few years old now but it still has 16gb of ram and an SSD and a Titan X.  No where near top of the line any more but its plenty for MM.  Id guess its something to do with one of the ally or opfor units making it hang.

**Comment by Thom293** [comment]
See EDIT2: Related to the Leadership bonus units.

**Comment by Thom293** [comment]
And perhaps helpful, but I just ran a battle that was 24 vs 45 and had no issues.  I dont think its a system resource issue.

**Comment by kernhe** [comment]
Bug seems to be related to #3135 

I sometimes have the same issue and when I start the game a second time from mhq there is no issue.

**Comment by Thom293** [comment]
Looks like at least the 2nd screenshot was caused by MM bug 2667.

**Comment by IllianiCBT** [comment]
I'm not familiar with any reports of this happening over the last couple years so I think we resolved it. Closing.

As always, if the issue reappears feel free to reopen the case. :)
